roachtest: schemachange/index/tpcc/w=100 failed

SHA: https://github.com/cockroachdb/cockroach/commits/57e825a7940495b67e0cc7213a5fabc24e12be0e

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=schemachange/index/tpcc/w=100 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1176948&tab=buildLog

```
The test failed on master:
	test.go:1202: test timed out (30m0s)
	cluster.go:1251,tpcc.go:138,schemachange.go:310,test.go:1214: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1176948-schemachange-index-tpcc-w-100:5 -- ./workload check tpcc --warehouses=100 {pgurl:1} returned:
		stderr:
		
		stdout:
		I190314 14:01:51.325134 1 workload/tpcc/tpcc.go:291  check 3.3.2.1 took 358.616857ms
		I190314 14:01:54.595379 1 workload/tpcc/tpcc.go:291  check 3.3.2.2 took 3.270139023s
		I190314 14:01:55.345421 1 workload/tpcc/tpcc.go:291  check 3.3.2.3 took 749.965545ms
		I190314 14:02:18.857901 1 workload/tpcc/tpcc.go:291  check 3.3.2.4 took 23.51238816s
		I190314 14:02:25.166335 1 workload/tpcc/tpcc.go:291  check 3.3.2.5 took 6.308355498s
		I190314 14:02:57.935942 1 workload/tpcc/tpcc.go:291  check 3.3.2.7 took 32.769537106s
		I190314 14:03:02.368606 1 workload/tpcc/tpcc.go:291  check 3.3.2.8 took 4.432602026s
		: signal: killed
	test.go:978,asm_amd64.s:523,panic.go:513,log.go:219,test.go:1174,asm_amd64.s:522,panic.go:397,test.go:774,test.go:760,cluster.go:1251,tpcc.go:138,schemachange.go:310,test.go:1214: write /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20190314-1176948/schemachange/index/tpcc/w=100/test.log: file already closed

```

SHA: https://github.com/cockroachdb/cockroach/commits/04ef15974085e14f758b20c552a84052eac9fa2b

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=schemachange/index/tpcc/w=100 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1180753&tab=buildLog

```
The test failed on master:
	schemachange.go:446,schemachange.go:314,cluster.go:1605,errgroup.go:57: read tcp 172.17.0.2:59392->35.196.208.131:26257: read: connection reset by peer
	cluster.go:1267,tpcc.go:130,cluster.go:1605,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1180753-schemachange-index-tpcc-w-100:5 -- ./workload run tpcc --warehouses=100 --histograms=logs/stats.json --wait=false --tolerate-errors --ramp=5m0s --duration=15m0s {pgurl:1-4} returned:
		stderr:
		
		stdout:
		0.0      0.0 delivery
		  13m16s    39606            0.0            0.8      0.0      0.0      0.0      0.0 newOrder
		  13m16s    39606            0.0            0.1      0.0      0.0      0.0      0.0 orderStatus
		  13m16s    39606            0.0            0.7      0.0      0.0      0.0      0.0 payment
		  13m16s    39606            0.0            0.0      0.0      0.0      0.0      0.0 stockLevel
		E190316 14:38:43.637780 1 workload/cli/run.go:420  error in payment: EOF
		_elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		  13m17s    62522            0.0            0.1      0.0      0.0      0.0      0.0 delivery
		  13m17s    62522            0.0            0.8      0.0      0.0      0.0      0.0 newOrder
		  13m17s    62522            0.0            0.1      0.0      0.0      0.0      0.0 orderStatus
		  13m17s    62522            0.0            0.7      0.0      0.0      0.0      0.0 payment
		  13m17s    62522            0.0            0.0      0.0      0.0      0.0      0.0 stockLevel
		: signal: killed
	cluster.go:1626,tpcc.go:140,schemachange.go:310,test.go:1214: unexpected node event: 3: dead

```

--- a/pkg/sql/schema_changer.go
+++ b/pkg/sql/schema_changer.go
@@ -278,8 +278,7 @@ func (sc *SchemaChanger) findTableWithLease(
 	}
 	if *tableDesc.Lease != lease {
 		log.Errorf(ctx, "table: %d has lease: %v, expected: %v", sc.tableID, tableDesc.Lease, lease)
-		return nil, pgerror.NewAssertionErrorWithWrappedErrf(errExpiredSchemaChangeLease,
-			"table: %d has lease: %v, expected: %v", log.Safe(sc.tableID), log.Safe(tableDesc.Lease), log.Safe(lease))
+		return nil, errExpiredSchemaChangeLease
 	}
 	return tableDesc, nil
 }

